instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/OkHttpTelemetry.java

```diff
@@ -67,8 +75,14 @@ public Interceptor newInterceptor() {
    */
   public Call.Factory newCallFactory(OkHttpClient baseClient) {
     OkHttpClient.Builder builder = baseClient.newBuilder();
-    // add our interceptor before other interceptors
-    builder.interceptors().add(0, newInterceptor());
+    // add our interceptors before other interceptors
+    builder.interceptors().add(0, new ContextInterceptor());
+    if (emitEncompassingSpan) {
+      builder.interceptors().add(1, new EncompassingSpanInterceptor());
+    } else if (createSpanForConnectionErrors) {
+      builder.interceptors().add(1, new ConnectionErrorSpanInterceptor(instrumenter));
+    }
```

Comment:
@trask @lmolkova I added two different ways of handling the connection stuff to this PR:
* the first one is the wrapper span
* the second one is a minor hack that only creates an HTTP span when there was an error thrown and no attempt to send an HTTP request was made - this way of handling things might not need any new spec (?)

Reply:
There are some example traces for both of these in the `OkHttp3Test` class